tsl/test/expected/continuous_aggs_invalidation.out

```diff
@@ -376,10 +458,53 @@ SELECT materialization_id AS cagg_id,
        greatest_modified_value AS end
        FROM _timescaledb_catalog.continuous_aggs_materialization_invalidation_log
        ORDER BY 1,2,3;
- cagg_id | start | end 
----------+-------+-----
-       3 |    60 | 100
-       4 |     1 |  19
-       4 |    60 | 100
-(3 rows)
+ cagg_id |        start         |         end         
+---------+----------------------+---------------------
+       3 | -9223372036854775808 |                  -1
+       3 |                   60 | 9223372036854775807
+       4 | -9223372036854775808 |                  19
+       4 |                   60 | 9223372036854775807
+       5 | -9223372036854775808 |                  -1
+       5 |                   30 | 9223372036854775807
+(6 rows)
+
+-- Refresh with two non-overlapping invalidations
+INSERT INTO conditions VALUES (20, 1, 23.4), (25, 1, 23.4);
+INSERT INTO conditions VALUES (30, 1, 23.4), (46, 1, 23.4);
+CALL refresh_continuous_aggregate('cond_10', 1, 60);
+SELECT materialization_id AS cagg_id,
+       lowest_modified_value AS start,
+       greatest_modified_value AS end
+       FROM _timescaledb_catalog.continuous_aggs_materialization_invalidation_log
+       ORDER BY 1,2,3;
+ cagg_id |        start         |         end         
+---------+----------------------+---------------------
+       3 | -9223372036854775808 |                  -1
+       3 |                   60 | 9223372036854775807
+       4 | -9223372036854775808 |                  19
+       4 |                   20 |                  25
+       4 |                   30 |                  46
+       4 |                   60 | 9223372036854775807
+       5 | -9223372036854775808 |                  -1
+       5 |                   30 | 9223372036854775807
+(8 rows)
+
+-- Test max refresh window
+CALL refresh_continuous_aggregate('cond_10', NULL, NULL);
```

Comment:
Shouldn't it refresh entire window? I see that the invalidation log wasn't affected, which seems to be odd.

Reply:
If you compare the before and after, you'll see a change. In particular, cagg 3 (which is `cond_10`) has these entries before the refresh:

```
 3 | -9223372036854775808 |                  -1
 3 |                   60 | 9223372036854775807
```

then after the refresh:

```
 3 | -9223372036854775808 | -9223372036854775801
 3 |  9223372036854775800 |  9223372036854775807
```

which is -infinity to start of (max) window and then end of (max) window to +infinity

Reply:
Sorry, I missed it. Is this exactly the case you described on a meeting that corner cases don't match bucket and cannot process correctly?

Reply:
If you are referring to the case when you don't have a full first bucket, then yes. It turns out that date_trunc also generates an error if a value fall into a bucket that is no complete. Therefore, I compute the max bucketed window in case of NULL, NULL input to refresh. Also not that there's really no definition of "infinity" for integer columns like there is for timestamps.